### DISCUSSION TITLE
Need more feedback about what is being reset when ALT-R hotkey is used from commit window

### DISCUSSION THREAD

**Discussion by wheezil** [gitextensions/gitextensions#8373] [General]
## Current behaviour

If recently had an unpleasant surprise.  I had learned that I could use ALT-R from the commit window to shorten the number of clicks/keystrokes to reset changes on selected files.  This works splendidly, however I found to my surprise that it reset ALL changes.  And so I did.  Oh, several days work lost!  And now I feel pretty dumb.  

## Expected behaviour

When ALT-R is pressed, or right-click followed by R, the feedback should contain some indication of what is about to happen, such as:

- Reset selected change lines?
- Reset all changes in file AwesomeContainer.java?
- Reset all changes to twenty files?
- ARE YOU SURE YOU WANT TO RESET CHANGES TO ALL STAGED FILES?

## Steps to reproduce

The behavior is arguably "as designed", so it is hard to call it a bug.  But since the consequences are dire, it may be worth calling a bug.  To reproduce, simply hit ALT-R and note that the confirmation dialog is exactly the same no mater what you are about to do.

## Did this work in previous version of GitExtensions

I think it has always worked like this.

## Environment

- Git Extensions 3.4.2.9737
- Build 082cfd97d09c47346638b1873f5c8cf6ed139382
- Git 2.20.1.windows.1 (recommended: 2.25.1 or later)
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4180.0
- DPI 96dpi (no scaling)

## Diagnostics

    ❤️ Git Extensions? Please consider supporting our collective:
    👉 https://opencollective.com/gitextensions

Sure I will support the effort, you guys seem really on the ball.  Thanks!

**Comment by gerhardol** [comment]
Converting to a discussion

See Settings-Hotkeys for the internal shortkeys.
There are few hotkeys with Alt, but there are standard Windows handling; if you press Alt you see underscores for shortcuts.

There is no direct Alt-R hotkey. In FormCommit, the R hotkey is default for resetting selected files and Alt, R to reset all files (for the button, can be hidden in settings).
All of the actions display a popup. (Some confirmations can be hidden, not this.)

Other: Commit often, then squash changes.

![image](https://user-images.githubusercontent.com/6248932/89218571-9780c780-d5ce-11ea-8a21-a1dc7e3d2d44.png)

**Reply by RussKie** [reply]
> There is no direct Alt-R hotkey.

It appears, there is. If no files are selected, <kbd>ALT+R</kbd> invokes "reset" sequence. 
If there is one or more items selected - [Reset] button is selected by default (❗).

**Comment by RussKie** [comment]
> When ALT-R is pressed, or right-click followed by R, the feedback should contain some indication of what is about to happen, such as:
> 
> * Reset selected change lines?
> * Reset all changes in file AwesomeContainer.java?
> * Reset all changes to twenty files?
> * ARE YOU SURE YOU WANT TO RESET CHANGES TO ALL STAGED FILES?

I empathise with your situation. But we do ask you whether you wish to reset files by showing the confirmation dialog:
![image](https://user-images.githubusercontent.com/4403806/89725022-0dad8000-da4e-11ea-9b5d-f7aee193582c.png)

Pressing <kbd>R</kbd> again invokes "Reset" option.

@gitextensions/git-extensions-team I think it is worth while considering updating the mnemonics on the [Reset] button to `R&eset`

**Reply by wheezil** [reply]
Yes I understand and appreciate the confirmation dialog.  What I would like is for the confirmation dialog to display what is about to happen.  As it is, the same message is displayed whether you are resetting one file, three files, or all files.  Ideally it would contain the names of the files being acted upon, up to a reasonable limit.  I realize that this comes down to "help us avoid dumb mistakes", but dumb mistakes seem to be a defining feature of my life ;-)
